File: core/worker/analysis/infrastructure/adapters/testdata/fixture-csharp/Data/UserRepository.cs
using System.Collections.Generic;
using Acme.Models;

namespace Acme.Data;

public class UserRepository
{
    private static int instanceCount;
    private readonly Dictionary<int, User> store = new();

    public User Find(int id) => store.TryGetValue(id, out var u) ? u : null;

    public void Save(User user) => store[user.Id] = user;
}

File: core/worker/analysis/infrastructure/adapters/testdata/fixture-java/src/main/java/com/acme/repo/UserRepository.java
package com.acme.repo;

import com.acme.model.User;
import java.util.HashMap;
import java.util.Map;

public class UserRepository {
    private static int instanceCount;
    private final Map<Integer, User> store = new HashMap<>();

    public User find(int id) {
        return store.get(id);
    }

    public void save(User user) {
        store.put(user.getId(), user);
    }
}
